cmd/pollexample/main.go

```diff
@@ -1,19 +1,34 @@
 package main
 
 import (
+	"bytes"
 	"context"
+	"io"
 	"log"
 	"os"
 
 	"github.com/sourcegraph/run"
 )
 
 func main() {
-	ctx := context.Background()
+	ctx, cancel := context.WithCancel(context.Background())
 
 	// Demonstrate that output streams live!
 	cmd := run.Bash(ctx, `for i in {1..10}; do echo -n "This is a test in loop $i "; date ; sleep 1; done`)
-	if err := cmd.Run().Stream(os.Stdout); err != nil {
+	if err := cmd.Run().
+		Filter(func(ctx context.Context, line []byte, dst io.Writer) (int, error) {
+			if bytes.Contains(line, []byte("loop 3")) {
+				defer cancel() // Interrupt parent context here
```

Comment:
I'm unsure to understand what we're trying to demonstrate here. We want to showcase that we're streaming the output, so both canceling and echoing the output in line 23 would achieve that. 

The output we're getting makes sense, but the intention around the cancel function call is a bit confusing for the potential reader. I would add a comment on line 23 saying that we're never going to print this because of the cancel just above to clarify for the reader. 

```
~/work/run/cmd/pollexample  linefilter-v2 $ go run ./...
go: downloading github.com/djherbis/buffer v1.2.0
go: downloading github.com/djherbis/nio/v3 v3.0.1
This is a test in loop 1 Tue May 17 09:58:48 CEST 2022
This is a test in loop 2 Tue May 17 09:58:49 CEST 2022
2022/05/17 09:58:51 signal: killed
exit status 1
```